naga/src/front/spv/mod.rs

```diff
@@ -5670,6 +5670,8 @@ impl<I: Iterator<Item = u32>> Frontend<I> {
                     format: map_image_format(format)?,
                     access: crate::StorageAccess::default(),
                 }
+            } else if is_sampled == 2 {
```

Comment:
Shouldn't this be checking that the format is not 0 (i.e. `ImageFormat::Unknown`) instead?

Reply:
At this point, format is already zero (non-zero handled by previous else if case)